### ISSUE TITLE
[Feature]: LoRA support for LLama 3.2 Vision Models

### ISSUE THREAD

**Issue by Gladiator07** [vllm-project/vllm#10824]
### 🚀 The feature, motivation and pitch

I tried loading finetuned LoRA for llama 3.2 11B vision instruct using the serve command (OpenAI client) and get this error message. 

```shell
ERROR 12-02 00:02:57 engine.py:366] NotImplementedError: LoRA is currently not currently supported with encoder/decoder models.
```

I finetuned the LoRA using unsloth's implementation. It would be great if we can have the support for LoRA for multimodal models as our team wants to use multiple LoRAs and merging the LoRA adapters to original model weights is not feasible for us. We are short on time for this project and as far as I can tell no other framework supports LoRA in this way. Also we need outlines for structured generation so vLLM (being the most user friendly, stable and mature framework ) is our best bet now. Can we get a timeline when will this be supported ? Also are there any workarounds possible until this feature is officially supported ?

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by DarkLight1337** [comment]
Do you have LoRA weights on the multimodal encoder or just the text encoder? The latter is more feasible but even in that case, we currently don't have any existing enc-dec models that use LoRA yet. @jeejeelee any thoughts on this?

**Comment by jeejeelee** [comment]
There are plans to support this, but it's not a high priority. It may take 1-2 months.

**Comment by DarkLight1337** [comment]
> There are plans to support this, but it's not a high priority. It may take 1-2 months.

Are you referring to encoder-decoder support or multimodal support?

**Comment by jeejeelee** [comment]
> > There are plans to support this, but it's not a high priority. It may take 1-2 months.
> 
> Are you referring to encoder-decoder support or multimodal support?

encoder-decoder.  The priority for multimodal should be before encoder-decoder

**Comment by Gladiator07** [comment]
> Do you have LoRA weights on the multimodal encoder or just the text decoder? The latter is more feasible but even in that case, we currently don't have any existing enc-dec models that use LoRA yet. @jeejeelee any thoughts on this?

I have LoRA weights for both encoder and text decoder. If the text decoder is feasible I will not add LoRA to vision encoder if it matches our evaluation criteria. Can you guide how to just use the text decoder LoRA ?

**Comment by jeejeelee** [comment]
For the `mllama` model, it currently doesn't support lora because it's an encoder-decoder multimodal model. Other models, such as `Idefics3`, support text decoder LoRA

**Comment by jeejeelee** [comment]
> Qwen2VL

Yeah, it should be noted that when training lora, it can only be added to the text decoder.
